Makefile

```diff
@@ -167,32 +165,86 @@ check-copyright:
 ###############################################################################
 # Release targets 
 ###############################################################################
-release: clean
-ifndef VERSION
-	$(error VERSION is undefined - run using make release VERSION=vX.Y.Z)
-endif
+## Produces a git tag for the release.
+release-tag: release-prereqs
 	git tag $(VERSION)
-	$(MAKE) image.created
+
+## Produces a clean build of release artifacts at the specified version.
+release-build: release-prereqs clean
+# Check that the correct code is checked out.
+ifneq ($(VERSION), $(GIT_VERSION))
+	$(error Attempt to build $(VERSION) from $(GIT_VERSION))
+endif
+
+	$(MAKE) image
 	docker tag $(CONTAINER_NAME) $(CONTAINER_NAME):$(VERSION)
 	docker tag $(CONTAINER_NAME) quay.io/$(CONTAINER_NAME):$(VERSION)
 
 	# Generate the `latest` images.
 	docker tag $(CONTAINER_NAME) quay.io/$(CONTAINER_NAME):latest
 
-# Ensure reported version is correct.
+	# Ensure reported version is correct.
 	if ! docker run $(CONTAINER_NAME):$(VERSION) -v | grep '^$(VERSION)$$'; then echo "Reported version:" `docker run $(CONTAINER_NAME):$(VERSION) version` "\nExpected version: $(VERSION)"; false; else echo "\nVersion check passed\n"; fi
```

Comment:
bit of duplication here with the target below, since this one-liner is fiddly I suggest making a new `make check-image-version` or something